google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/ChangeStreamRecordAdapter.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.data.v2.models;
+
+import com.google.api.core.InternalApi;
+import com.google.bigtable.v2.ReadChangeStreamResponse;
+import com.google.cloud.bigtable.data.v2.models.Range.TimestampRange;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.Timestamp;
+import javax.annotation.Nonnull;
+
+/**
+ * An extension point that allows end users to plug in a custom implementation of logical change
+ * stream records. This is useful in cases where the user would like to apply advanced client side
+ * filtering(for example, only keep DeleteFamily in the mutations). This adapter acts like a factory
+ * for a SAX style change stream record builder.
+ */
+public interface ChangeStreamRecordAdapter<ChangeStreamRecordT> {
```

Comment:
Should this be InternalExtensionOnly?

Reply:
Currently it's only our java client(i.e. merging callable..). I mimic RowsAdapter just in case in the future we want to allow customized filtering on ChangeStreamMutations